python/ray/rllib/a3c/common.py

```diff
@@ -5,31 +5,59 @@
 import numpy as np
 import scipy.signal
 from collections import namedtuple
+from ray.rllib.ppo.filter import MeanStdFilter, NoFilter
 
 
 def discount(x, gamma):
     return scipy.signal.lfilter([1], [1, -gamma], x[::-1], axis=0)[::-1]
 
 
-def process_rollout(rollout, gamma, lambda_=1.0):
-    """Given a rollout, compute its returns and the advantage."""
-    batch_si = np.asarray(rollout.states)
-    batch_a = np.asarray(rollout.actions)
-    rewards = np.asarray(rollout.rewards)
-    vpred_t = np.asarray(rollout.values + [rollout.r])
+def process_rollout(rollout, reward_filter, gamma, lambda_=1.0):
+    """Given a rollout, compute its returns and the advantage.
 
-    rewards_plus_v = np.asarray(rollout.rewards + [rollout.r])
+    TODO(rliaw): generalize this"""
+    batch_si = np.asarray(rollout.data["state"])
+    batch_a = np.asarray(rollout.data["action"])
+    rewards = np.asarray(rollout.data["reward"])
+    vpred_t = np.asarray(rollout.data["value"] + [rollout.last_r])
+
+    rewards_plus_v = np.asarray(rollout.data["reward"] + [rollout.last_r])
     batch_r = discount(rewards_plus_v, gamma)[:-1]
     delta_t = rewards + gamma * vpred_t[1:] - vpred_t[:-1]
     # This formula for the advantage comes "Generalized Advantage Estimation":
     # https://arxiv.org/abs/1506.02438
     batch_adv = discount(delta_t, gamma * lambda_)
+    for i in range(batch_adv.shape[0]):
+        batch_adv[i] = reward_filter(batch_adv[i])
 
-    features = rollout.features[0]
-    return Batch(batch_si, batch_a, batch_adv, batch_r, rollout.terminal,
+    features = rollout.data["features"][0]
+    return Batch(batch_si, batch_a, batch_adv, batch_r, rollout.is_terminal(),
                  features)
 
 
+def get_filter(filter_config, shape):
```

Comment:
I think it's time we moved PPO's filters to a common place (both DQN+A3C use it now). How about `rllib/util`?